internal/safefile/safeopen.go

```diff
@@ -339,6 +339,35 @@ func MkdirRelative(path string, root *os.File) error {
 	return err
 }
 
+// MkdirAllRelative creates each directory in the path relative to a root, failing if
+// any existing intermediate path components are reparse points.
+func MkdirAllRelative(path string, root *os.File) error {
```

Comment:
Nit: mkdir on windows creates subfolders automatically; on linux `-p` serves that purpose. Does that help simplify all this?

Reply:
`safefile.MkdirRelative` does that already, and we call it from here. This is for the part where the subfolders _already_ exist.